mne/time_frequency/tfr.py

```diff
@@ -1250,34 +1254,44 @@ def _plot(self, picks=None, baseline=None, mode='mean', tmin=None,
 
         data = tfr.data
         n_picks = len(tfr.ch_names) if combine is None else 1
+
+        # combine picks
         if combine == 'mean':
             data = data.mean(axis=0, keepdims=True)
         elif combine == 'rms':
             data = np.sqrt((data ** 2).mean(axis=0, keepdims=True))
         elif combine is not None:
             raise ValueError('combine must be None, mean or rms.')
 
-        if isinstance(axes, list) or isinstance(axes, np.ndarray):
-            if len(axes) != n_picks:
-                raise RuntimeError('There must be an axes for each picked '
-                                   'channel.')
-
+        # figure overhead
+        # set plot dimension
         tmin, tmax = tfr.times[[0, -1]]
         if vmax is None:
             vmax = np.abs(data).max()
         if vmin is None:
             vmin = -np.abs(data).max()
-        if isinstance(axes, plt.Axes):
-            axes = [axes]
 
+        # set colorbar
         cmap = _setup_cmap(cmap)
-        for idx in range(len(data)):
-            if axes is None:
-                fig = plt.figure()
-                ax = fig.add_subplot(111)
-            else:
-                ax = axes[idx]
-                fig = ax.get_figure()
+
+        # make sure there are as many axes as there will be channels to plot
+        if isinstance(axes, list) or isinstance(axes, np.ndarray):
+            figs_and_axes = [(ax.get_figure(), ax) for ax in axes]
+        elif isinstance(axes, plt.Axes):
+            figs_and_axes = [(ax.get_figure(), ax) for ax in [axes]]
+        elif axes is None:
```

Comment:
if `axes` is None and `combine` is not None then there should be only one figure

Reply:
Yes, that would happen here as well. Earlier (new line 1256), `n_picks` is set to `1` if `combine` is not `None`. So that here only a single figure would be created.

Reply:
ok!